Description: Add an helper to detect current desktop
 We will differentiate some behavior depending on current desktop. Add an
 helper to centralize the current desktop detection.
Forwarded: not-needed
Origin: ubuntu
Author: didrocks@ubuntu.com
Index: gnome-shell-3.24.3/js/misc/desktop.js
===================================================================
--- /dev/null
+++ gnome-shell-3.24.3/js/misc/desktop.js
@@ -0,0 +1,39 @@
+// -*- mode: js; js-indent-level: 4; indent-tabs-mode: nil -*-
+
+const GLib = imports.gi.GLib;
+
+// current desktop doesn't change unless we restart the shell or control
+// the env variable. It's safe to cache matching result
+let _currentDesktopsMatches = {};
+
+// is:
+// @name: desktop string you want to assert if it matches the current desktop env
+//
+// The function examples XDG_CURRENT_DESKTOP and return if the current desktop
+// is part of that desktop string.
+//
+// Return value: if the environment isn't set or doesn't match, return False
+// otherwise, return True.
+function is(name) {
+
+    if (_currentDesktopsMatches[name] !== undefined) {
+        return _currentDesktopsMatches[name];
+    }
+
+    let desktopsEnv = GLib.getenv('XDG_CURRENT_DESKTOP');
+    if (!desktopsEnv) {
+        _currentDesktopsMatches[name] = false;
+        return false;
+    }
+
+    let desktops = desktopsEnv.split(":");
+    for (let i = 0; i < desktops.length; i++) {
+        if (desktops[i] === name) {
+            _currentDesktopsMatches[name] = true;
+            return true;
+        }
+    }
+
+    _currentDesktopsMatches[name] = false;
+    return false;
+}
Index: gnome-shell-3.24.3/js/js-resources.gresource.xml
===================================================================
--- gnome-shell-3.24.3.orig/js/js-resources.gresource.xml
+++ gnome-shell-3.24.3/js/js-resources.gresource.xml
@@ -13,6 +13,7 @@
 
     <file>misc/config.js</file>
     <file>misc/extensionUtils.js</file>
+    <file>misc/desktop.js</file>
     <file>misc/fileUtils.js</file>
     <file>misc/gnomeSession.js</file>
     <file>misc/history.js</file>
